pkg/rewards/claim.go

```diff
@@ -75,13 +80,142 @@ func getClaimFlags() []cli.Flag {
+	var accounts []merkletree.MerkleTree
+
+	for _, claimConfig := range claimConfigs {
+		earnerAddr := gethcommon.HexToAddress(claimConfig.EarnerAddress)
+
+		var tokenAddrs []gethcommon.Address
+
+		// Empty token addresses list will create a claim for all tokens claimable
+		// by the earner address.
+		if len(claimConfig.TokenAddresses) != 0 {
+			for _, addr := range claimConfig.TokenAddresses {
+				tokenAddrs = append(tokenAddrs, gethcommon.HexToAddress(addr))
+			}
+		}
+
+		elClaim, claim, account, err := generateClaimPayload(
+			ctx,
+			rootIndex,
+			proofData,
+			elReader,
+			logger,
+			earnerAddr,
+			tokenAddrs,
+		)
+
+		if err != nil {
+			logger.Warnf("Failed to process claim for earner %s: %v", earnerAddr.String(), err)
+			continue
+		}
+
+		elClaims = append(elClaims, *elClaim)
+		claims = append(claims, *claim)
+		accounts = append(accounts, *account)
+
+	}
+
+	err = broadcastClaims(config, ethClient, logger, p, ctx, &elClaims, &claims, &accounts)
+	return err
+}
+
+func generateClaimPayload(
+	ctx context.Context,
+	rootIndex uint32,
+	proofData *proofDataFetcher.RewardProofData,
+	elReader *elcontracts.ChainReader,
+	logger logging.Logger,
+	earnerAddress gethcommon.Address,
+	tokenAddresses []gethcommon.Address,
+) (*rewardscoordinator.IRewardsCoordinatorRewardsMerkleClaim, *contractrewardscoordinator.IRewardsCoordinatorRewardsMerkleClaim, *merkletree.MerkleTree, error) {
+
+	claimableTokensOrderMap, present := proofData.Distribution.GetTokensForEarner(earnerAddress)
+	if !present {
+		return nil, nil, nil, errors.New("no tokens claimable by earner")
+	}
+
+	claimableTokensMap := getTokensToClaim(claimableTokensOrderMap, tokenAddresses)
+
+	claimableTokens, err := filterClaimableTokens(ctx, elReader, earnerAddress, claimableTokensMap)
+	if err != nil {
+		return nil, nil, nil, eigenSdkUtils.WrapError("failed to get claimable tokens", err)
+	}
+
+	cg := claimgen.NewClaimgen(proofData.Distribution)
+	accounts, claim, err := cg.GenerateClaimProofForEarner(
+		earnerAddress,
+		claimableTokens,
+		rootIndex,
+	)
+	if err != nil {
+		return nil, nil, nil, eigenSdkUtils.WrapError("failed to generate claim proof for earner", err)
+	}
+
+	elClaim := rewardscoordinator.IRewardsCoordinatorRewardsMerkleClaim{
+		RootIndex:       claim.RootIndex,
+		EarnerIndex:     claim.EarnerIndex,
+		EarnerTreeProof: claim.EarnerTreeProof,
+		EarnerLeaf: rewardscoordinator.IRewardsCoordinatorEarnerTreeMerkleLeaf{
+			Earner:          claim.EarnerLeaf.Earner,
+			EarnerTokenRoot: claim.EarnerLeaf.EarnerTokenRoot,
+		},
+		TokenIndices:    claim.TokenIndices,
+		TokenTreeProofs: claim.TokenTreeProofs,
+		TokenLeaves:     convertClaimTokenLeaves(claim.TokenLeaves),
+	}
+
+	logger.Infof("Validating claim proof for earner %s...", earnerAddress)
+	ok, err := elReader.CheckClaim(ctx, elClaim)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	if !ok {
+		return nil, nil, nil, errors.New("failed to validate claim")
+	}
```

Comment:
May not be relevant, but I'm confused why this has both an error and an `ok` check. Can we combine these?

Reply:
Introduced here, @shrimalmadhur  should I combine them? https://github.com/Layr-Labs/eigenlayer-cli/pull/213/files

Reply:
those are two different errors so as long as we can send the correct error. because err can be non nil if let's say rpc not working or something else. so if we can send correct error, feel free to combine

Reply:
Will keep as is given that context